include/circt/Dialect/SV/SVInOutOps.td

```diff
@@ -139,3 +140,38 @@ def ArrayIndexInOutOp
   let assemblyFormat =
     "$input`[`$index`]` attr-dict `:` type($input) `,` type($index)";
 }
+
+def IndexedPartSelectInOutOp : SVOp<"indexed_part_select_inout",
+                         [NoSideEffect, InferTypeOpInterface]> {
+  let summary = "Address several contiguous bits of an inout type (e.g. a wire"
+                " or inout port). This is an indexed part-select operator."
+                "The base is an integer expression and the width is an "
+                " integer constant. The bits start from base and the number "
+                "of bits selected is equal to width. If $decrement is true, "
+                " then part select decrements starting from $base."
+                "See SV Spec 11.5.1.";
+
+  let arguments = (ins InOutType:$input, HWIntegerType:$base, I32Attr:$width,
+                      UnitAttr:$decrement);
```

Comment:
I'm not sure what use-case you have in mind for this, but is there any reason we care about increment vs decrement syntax?  Can we just eliminate the decrement form to simplify the IR?

Reply:
I think just the increment would work, will discuss again with Lenharth to remove the decrement.